modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/wgs_linear_solver.h

```diff
@@ -13,14 +13,15 @@ namespace opensn
 {
 
 /// Linear Solver specialization for Within GroupSet (WGS) solves.
-class WGSLinearSolver : public LinearSolver
+class WGSLinearSolver : public PETScLinearSolver
```

Comment:
Should this be the inheritance structure here? What if `ClassicRichardson` is being used?

Reply:
I think so. Since `WGSLinearSolver` is a PETSc solver, it derives from `PETScLinearSolver`. Classic Richardson doesn't need any of the PETSc functionality, so it derives from `LinearSolver`.

Reply:
Right, that makes sense. I was misremembering some structure. Either a `WGSLinearSolver` or a `ClassicRichardson` solver gets initialized, the `ClassicRichardson` stuff is not embedded in `WGSLinearSolver`. Should we rename `ClassicRichardson` to `WGSClassicRichardson`?

Reply:
Yes, that would fit better with the naming scheme. I'll put up a PR for that.